test/usecase/login_use_case_test.dart

```diff
@@ -0,0 +1,56 @@
+import 'package:flutter_survey/api/exception/network_exceptions.dart';
+import 'package:flutter_survey/models/oauth_token.dart';
+import 'package:flutter_survey/usecase/base/base_use_case.dart';
+import 'package:flutter_survey/usecase/login_use_case.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
+
+import '../../test/mock/mock_data.mocks.dart';
+import '../fake/fake_shared_preference_helper.dart';
+
+void main() {
+  group('LoginUseCaseTest', () {
+    late MockOAuthRepository mockRepository;
+    late FakeSharedPreferencesHelper fakeSharePref;
+    late LoginUseCase loginUseCase;
+
+    setUp(() {
+      mockRepository = MockOAuthRepository();
+      fakeSharePref = FakeSharedPreferencesHelper();
+      loginUseCase = LoginUseCase(mockRepository, fakeSharePref);
+    });
+
+    test(
+        'When logging in with valid email and password, it returns Success result',
+        () async {
+      when(mockRepository.login(
+              email: anyNamed('email'), password: anyNamed('password')))
+          .thenAnswer((_) async => OAuthToken(
+                accessToken: "accessToken",
+                tokenType: "tokenType",
+                expiresIn: 111,
+                refreshToken: "refreshToken",
+              ));
+      final result = await loginUseCase
+          .call(LoginInput(email: 'email', password: 'password'));
+
+      expect(result, isA<Success>());
+    });
+
+    test(
+        'When logging in with incorrect email or password, it returns Failed result',
+        () async {
+      when(mockRepository.login(
+              email: anyNamed('email'), password: anyNamed('password')))
+          .thenAnswer((_) => Future.error(
```

Comment:
I think you can use thenThrow to avoid creating a Future.error

Reply:
I think moving forward we better decide to use try/catch or Future/then/error to be consistent. I found that I was wrong about using `func.then().error()` before too 😓  I'm checking if this is why the catch error by using `error()` block wasn't called

https://dart.dev/guides/libraries/futures-error-handling

![image](https://user-images.githubusercontent.com/13435717/137773188-941c97a1-25cc-416e-ba57-33e6fd9e06ec.png)

Reply:
they are the same imo 🤔  Future = async func {} . Await = then (). catch = catchError 🤷

Reply:
similar issue here: https://stackoverflow.com/questions/63493407/dart-future-catcherror-not-called-in-unit-test 😔

Reply:
omg, I thought you got why I can't apply `thenThrow` here already 🥲 We need to mock with Future here, the one in Repository is different.